### ISSUE TITLE
Pointer being freed was not allocated

### ISSUE THREAD

**Issue by larrypl** [gwsw/less#85]
I use `/@string` a lot in less (start searching from the beginning of the file). It _looks_ like the second time `/@string` is issued in a row, I get 
```
less(58799,0x10a970dc0) malloc: *** error for object 0x7fdc496040d0: pointer being freed was not allocated
less(58799,0x10a970dc0) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```
And because I was compiling less with posix2 regexes, I tried with the default (posix) regexes and got the same message.
Is this a bug in less or possibly a library I'm using?

This is on MacOS 10.15.7, compiling with gcc 10.2.0.

**Comment by gwsw** [comment]
I cannot reproduce this on MacOS with the posix regex. Can you confirm that this crashes for you on less-563, and if so, describe your steps to reproduce it?

**Comment by larrypl** [comment]
Sorry - yes, less 563

> $ less -V
less 563 (PCRE2 regular expressions)
Copyright (C) 1984-2020  Mark Nudelman

> less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: http://www.greenwoodsoftware.com/less

I do `less /etc/passwd` (for example), type in `/@foo<enter>/@bar<enter>` and get the malloc error.

This does _not_ fail when doing `less --help`. It also doesn't seem to fail when piping to less.

I have
```
$ echo $LESSOPEN
| /Users/larry/bin/lesspipe.sh %s
larryp-MBP:~ larry$ ls -l bin/lesspipe.sh
lrwxr-xr-x 1 larry staff 34 Sep 26 22:15 bin/lesspipe.sh -> ../../../opt/local/bin/lesspipe.sh
$ ls -lL bin/lesspipe.sh
-rwxr-xr-x 1 root wheel 34718 Aug 25 06:01 bin/lesspipe.sh
```

and

```
$ echo $LESS
SMFJWRciaj.4#.99h0y0 --mouse
```

However it still fails with `$ LESS= less /etc/passwd`. It succeeds with `$ LESS= LESSOPEN= less /etc/passwd` so it seems to be some interaction with lesspipe.sh.

Lesspipe is

>$ port info lesspipe
lesspipe @1.85 (textproc)

> Description:          lesspipe.sh is an input filter for the pager less as described in its man page. The script runs under a ksh compliant shell (ksh, bash, zsh) and allows to view files with binary content, compressed files, archives and files contained in archives.
                      A large and growing number of formats are supported both as plain and compressed files using compress, gzip, bzip2 or zip.
Homepage:             https://github.com/wofr06/lesspipe

> Platforms:            darwin
License:              GPL-2
Maintainers:          none


Less fails similarly with brew's lesspipe.sh, which is also version 1.85, and with LESS_ADVANCED_PREPROCESSOR=1.

**Comment by larrypl** [comment]
It probably goes without saying that it doesn't fail with `less -L /etc/passwd`.

**Comment by larrypl** [comment]
I've opened [lesspipe issue 47](https://github.com/wofr06/lesspipe/issues/47) on this also, in case they can determine the exact conditions required to re-create this.

**Comment by larrypl** [comment]
No comment on this issue? Less is clearly doing _something_ wrong, at least not handling an error condition gracefully.

**Comment by gwsw** [comment]
Sorry, I've been a bit busy lately but I should have updated this. I was able to reproduce the problem, and I was able to reproduce it using the simple example script from the less man page. So I think the problem is in less's handling of LESSOPEN and is not related to anything that any particular lesspipe script is doing. I do not yet know the root cause and I am still investigating.

**Comment by larrypl** [comment]
Thanks very much for the update! Good to know you were able to reproduce it. I think I'll close the case on less pipe.

**Comment by gwsw** [comment]
Fixed in 6c6bee2ffb0711e86f310f5c592589a7164a0768.
